onnxruntime/core/providers/cpu/math/matmul.cc

```diff
@@ -134,7 +172,24 @@ Status MatMul<float>::PrePack(const Tensor& tensor, int input_idx, /*out*/ Alloc
   // only pack Matrix B
   if (input_idx == 1) {
     size_t packed_b_size;
-    is_packed = GemmPackBFp32(alloc, tensor, trans_b_attr_ != 0, packed_b_, packed_b_size, b_shape_);
+#if defined(__aarch64__) && defined(__linux__)
+    size_t dim1 = 0;
+    size_t dim2 = 0;
+    TensorShape b_shape = tensor.Shape();
+
+    if (b_shape.NumDimensions() == 2) {
+      dim1 = static_cast<size_t>(b_shape[0]);
+      dim2 = static_cast<size_t>(b_shape[1]);
+    }
+
+    if (use_fastmath_mode_ && (trans_b_attr_ == 0) && ((dim1 * dim2) >= kFastMathModeKernelsizeThreshold)) {
+      is_packed = GemmPackBBfloat16(alloc, tensor, trans_b_attr_ != 0, packed_b_, packed_b_size, b_shape_);
+    } else
```

Comment:
what if we are on an ARM64 chip that does not support bf16? Then the GemmPackBBfloat16 should fail. But then we should fall back to packing fp32. This logic does not seem right to me.

Reply:
`use_fastmath_mode_` is set to true only for the platforms that have bf16 support and the mode is explicitly enabled through session options.

```
#if defined(__aarch64__) && defined(__linux__)
    auto config_ops = info.GetConfigOptions().GetConfigEntry(kOrtSessionOptionsGemmFastMathMode);
    use_fastmath_mode_ = (config_ops == "1") && CPUIDInfo::GetCPUIDInfo().HasArmNeon_BF16();
#endif
```